console/src/main/java/com/arcadedb/console/Console.java

```diff
@@ -312,6 +314,28 @@ private void executeClose() {
     }
   }
 
+  private void executeList(final String line) {
+    final String url = line.substring("list".length()).trim();
+
+    final String[] urlParts = url.split(" ");
+
+    outputLine("Databases:");
+    if (urlParts[0].startsWith(REMOTE_PREFIX)) {
+      connectToRemoteServer(url, false);
```

Comment:
I think this could cause a NPE if the remote connection is null. I think it should be checked that is not null in the same if above. WDYT?

Reply:
I thought about that, and figured that if `urlParts[0]` is empty, then the conditional evaluates to the alternate (as then a local connection is requested, i.e. `> list`) and thus the consequent is not reachable in this case. But I may have overlooked something?

Reply:
You could already have the connection taken from the server. Think to the case you connect to the server first, and then you're asking for the list of databases hosted on that server:

```
connect remote:... user password
list databases
```

(By the way, is "list database" better than just "list"?)

Reply:
Oh, I think I see your point: You connect to a database on a remote server first, then ask for a list of databases, then for try to interact with the database again. In this case, listing the databases acts like a `close`, as it overwrites  the `remoteDatabase property`. I agree, this is not intuitive. How about having a temporary storing the current `RemoteDatabase` instance during the `list` and restoring it at the end? This way the user is not forced to `close` a connection for a simple server command.

BTW: I agree `list databases` is a better name.

Reply:
What if we remove the URL from `list databases`? It would behave differently based on if you're connected to a remote server or not = local.

Example with remote:

```
connect remote:... user passed; list databases
```

Example with local:

```
list databases
```

Reply:
Hmm, `connect` currently requires a database, as (in my understanding) it connects to a database not "just" to a server. So IMHO, just using `list databases` would mean one couldn't list databases on a server without (known) databases. This would then mean `connect` needs to be changed to "connect to a server"?

Reply:
You're right, forget my last comment. The PR is good to me to be merged.